monai/data/image_reader.py

```diff
@@ -235,11 +236,28 @@ def _get_array_data(self, img) -> np.ndarray:
         """
         Get the raw array data of the image, converted to Numpy array.
 
+        Following PyTorch conventions, the returned array data has contiguous channels,
+        e.g. for an RGB image, all red channel image pixels are contiguous in memory.
+        The first axis of the returned array is the channel axis.
+
         Args:
             img: a ITK image object loaded from a image file.
 
         """
-        return itk.array_view_from_image(img, keep_axes=False)
+        channels = img.GetNumberOfComponentsPerPixel()
+        if channels == 1:
+            return itk.array_view_from_image(img, keep_axes=False)
+        else:
+            # The memory layout of itk.Image has all pixel's channels adjacent
+            # in memory, i.e. R1G1B1R2G2B2R3G3B3. For PyTorch/MONAI, we need
+            # channels to be contiguous, i.e. R1R2R3G1G2G3B1B2B3.
+            arr = itk.array_view_from_image(img, keep_axes=False)
+            dest = list(range(img.ndim))
+            source = dest.copy()
+            end = source.pop()
+            source.insert(0, end)
+            arr_contiguous_channels = np.moveaxis(arr, source, dest)
```

Comment:
Wouldn't `np.moveaxis(arr,-1,0)` have the same effect?

Reply:
I tried that at first, but it does not work as desired.

Reply:
My test is this, which yields something of the right shape but perhaps the array in memory isn't what we want:

```py
x=np.random.rand(10, 20, 3)
dest = list(range(x.ndim))  # [10, 20, 3]
source = dest.copy()
end = source.pop()
source.insert(0, end)
print(x.shape, np.moveaxis(x, source, dest).shape, np.moveaxis(x, -1, 0).shape)
>>> ((10, 20, 3), (3, 10, 20), (3, 10, 20))
```

Another way to shorten the code is construct `source` as `source = np.roll(dest, 1)`.